src/FileOnQ.Imaging.Raw/RawImageData.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace FileOnQ.Imaging.Raw
+{
+	// NOTE - 8/4/2021 - @ahoefling - This wrapper exists so we can add buffer pointers to allow further customization in IImageProcessors
+	public class RawImageData
+	{
+		public IntPtr LibRawData { get; set; }
+	}
```

Comment:
This object is a way to future-proof the API structure. Right now we are taking a big dependency on the native library LibRaw. By passing `RawImageData` to all the APIs in `IImageProcessor` it allows the downstream implementation to use `LibRaw` or another technology. In the future this model will change to include a pointer to the raw buffer which can then be manipulated. That will open up the door for GPU acceleration and other techniques that don't require LibRaw.

We can remove this and just take on the breaking change when it comes, but I think adding this structure sets us up for how it will look in the future

Reply:
If a true wrapper, should we avoid the name of LibRawData? Maybe just RawData?

Reply:
The idea for the future is to remove `LibRawData` or just leave it in while we take the dependency of `LibRaw`. In the future I would like to have all of the raw data pointers or managed data on the `RawImageData` object. Consider the model below
```c#
public class RawImageData
{
  IntPtr LibRawData { get; set; }
  byte* Buffer { get; set; }
  byte* Colors { get; set; }
  byte* Matrix { get; set; }
}
```

Eventually I would like to have all the data available for downstream extensions. But while we have a LibRaw dependency I think it is still useful to have the pointer to their data object. This will allow downstream projects to just invoke `LibRaw` APIs in their processor

If we remove LibRaw in the future, it is a breaking change no matter what

Reply:
Sounds good to me